tests/e2e/service_instances_test.go

```diff
@@ -180,4 +177,48 @@ var _ = Describe("Service Instances", func() {
 			})
 		})
 	})
+
+	Describe("List", func() {
+		var (
+			anotherSpaceGUID     string
+			anotherInstanceGUID  string
+			serviceInstancesList resourceList[resource]
+		)
+
+		BeforeEach(func() {
+			anotherSpaceGUID = createSpace(generateGUID("space1"), commonTestOrgGUID)
+			anotherInstanceGUID = createServiceInstance(anotherSpaceGUID, generateGUID("service-instance"), nil)
+		})
+
+		JustBeforeEach(func() {
+			serviceInstancesList = resourceList[resource]{}
+			httpResp, httpError = certClient.R().SetResult(&serviceInstancesList).Get("/v3/service_instances")
+		})
+
+		It("returns an empty list", func() {
```

Comment:
This test is failing on CI.

Reply:
I've been playing around with it and I think this test is very sensitive to test pollution from others that are running concurrently (or fail to clean up properly). I wonder if we should make this assertion more like this one that tests app listing:

https://github.com/cloudfoundry/korifi/blob/main/tests/e2e/apps_test.go#L69-L78

Instead of asserting on the absolute contents we could just check that the list contains a few service instances that we know should be in there. And we make a new space that we explicitly don't have access to and verify that the instances in that space aren't present.

Reply:
Thanks for spotting this @tcdowney 

I have applied your proposal so the test should be fine now.